File: AI.Core/OperatorsImplimentation/MetaOperators/DeleteChannel.cs
using System;
using System.Collections.Generic;
using AI.Core.ChannelsImplimentation;

namespace AI.Core.OperatorsImplimentation.MetaOperators
{
    class DeleteChannel : ChannelOperator
    {
        /// <summary>
        /// Delete Channel
        /// </summary>
        public DeleteChannel(Entity thisEntity)
            : base(thisEntity){}

        public override void Action()
        {
            try
            {
                Channel channel;
                var chktey = new ChannelTableKey(FromOperator, ToOperator);
                try
                {
                    channel = ThisEntity.ChannelsTable[chktey];
                }
                catch (KeyNotFoundException) {return;}

                FromOperator.ExitContacts[channel.FromContact.ContactNumber] = null;
                ToOperator.EnterContacts[channel.ToContact.ContactNumber] = null;
                ThisEntity.ChannelsTable[chktey] = null;
            }
            catch (ArgumentOutOfRangeException){}
        }
    }
}

File: NativeCore/DeleteChannel.cpp
#include "DeleteChannel.h"
#include "Channel.h"
#include "ChannelsTable.h"
#include "stdafx.h"

//C# TO C++ CONVERTER TODO TASK: The .NET System namespace is not available from native C++:
//using namespace System;
//C# TO C++ CONVERTER TODO TASK: The .NET System namespace is not available from native C++:
//using namespace System::Collections::Generic;
using namespace AI::Core::ChannelsImplimentation;
namespace AI
{
	namespace Core
	{
		namespace OperatorsImplimentation
		{
			namespace MetaOperators
			{

				DeleteChannel::DeleteChannel(Entity *thisEntity) : ChannelOperator(thisEntity)
				{
				}

				void DeleteChannel::Action()
				{
					try
					{
						Channel *channel;
						auto chktey = ChannelTableKey(getFromOperator(), getToOperator());
						try
						{
							channel = getThisEntity()->getChannelsTable()[chktey];
						}
						catch (KeyNotFoundException e1)
						{
							return;
						}

//C# TO C++ CONVERTER WARNING: C# to C++ Converter converted the original 'null' assignment to a call to 'delete', but you should review memory allocation of all pointer variables in the converted code:
						delete getFromOperator()->ExitContacts[channel->getFromContact().getContactNumber()];
//C# TO C++ CONVERTER WARNING: C# to C++ Converter converted the original 'null' assignment to a call to 'delete', but you should review memory allocation of all pointer variables in the converted code:
						delete getToOperator()->EnterContacts[channel->getToContact().getContactNumber()];
						getThisEntity()->getChannelsTable()[chktey] = nullptr;
					}
					catch (ArgumentOutOfRangeException e2)
					{
					}
				}
			}
		}
	}
}
